tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -39,12 +39,43 @@
 import time
 import logging
 import logging.config
+import inspect
 
 import rospkg
 from rospkg.environment import ROS_LOG_DIR
 
 class LoggingException(Exception): pass
 
+class RospyLogger(logging.getLoggerClass()):
+    def findCaller(self):
+        """
+        Find the stack frame of the caller so that we can note the source
+        file name, line number, and function name with class name if possible.
```

Comment:
While this is a good description of the function itself would it be better to document what this specific variation does (extending the function name with the class name if applicable)?